diff --git a/cube/CMakeLists.txt b/cube/CMakeLists.txt
index 4087351d..604283e9 100644
--- a/cube/CMakeLists.txt
+++ b/cube/CMakeLists.txt
@@ -228,21 +228,19 @@ include_directories(${CUBE_INCLUDE_DIRS})
 if(APPLE)
     include(macOS/cube/cube.cmake)
 elseif(NOT WIN32)
-    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
-        add_executable(vkcube
+    add_executable(vkcube
                        cube.c
                        ${PROJECT_SOURCE_DIR}/cube/cube.vert
                        ${PROJECT_SOURCE_DIR}/cube/cube.frag
                        cube.vert.inc
                        cube.frag.inc
                        ${OPTIONAL_WAYLAND_DATA_FILES})
-        target_link_libraries(vkcube Vulkan::Vulkan)
-        target_compile_definitions(vkcube PUBLIC ${CUBE_PLATFORM})
-        include(CheckLibraryExists)
-        CHECK_LIBRARY_EXISTS("rt" clock_gettime "" NEED_RT)
-        if (NEED_RT)
+    target_link_libraries(vkcube Vulkan::Vulkan)
+    target_compile_definitions(vkcube PUBLIC ${CUBE_PLATFORM})
+    include(CheckLibraryExists)
+    CHECK_LIBRARY_EXISTS("rt" clock_gettime "" NEED_RT)
+    if (NEED_RT)
             target_link_libraries(vkcube rt)
-        endif()
     endif()
 else()
     if(CMAKE_CL_64)
@@ -280,7 +278,6 @@ endif()
 if(APPLE)
     include(macOS/cubepp/cubepp.cmake)
 elseif(NOT WIN32)
-    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
         add_executable(vkcubepp
                        cube.cpp
                        ${PROJECT_SOURCE_DIR}/cube/cube.vert
@@ -290,7 +287,6 @@ elseif(NOT WIN32)
                        ${OPTIONAL_WAYLAND_DATA_FILES})
         target_link_libraries(vkcubepp Vulkan::Vulkan)
         target_compile_definitions(vkcubepp PUBLIC ${CUBE_PLATFORM})
-    endif()
 else()
     if(CMAKE_CL_64)
         set(LIB_DIR "Win64")
@@ -334,7 +330,6 @@ if(UNIX AND NOT APPLE) # i.e. Linux
             ${CMAKE_CURRENT_BINARY_DIR}/xdg-decoration-code.c
             ${CMAKE_CURRENT_BINARY_DIR}/xdg-decoration-client-header.h)
         include_directories(${CMAKE_CURRENT_BINARY_DIR})
-        if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
             add_executable(vkcube-wayland
                            cube.c
                            ${PROJECT_SOURCE_DIR}/cube/cube.vert
@@ -349,6 +344,5 @@ if(UNIX AND NOT APPLE) # i.e. Linux
             if (NEED_RT)
                 target_link_libraries(vkcube-wayland rt)
             endif()
-        endif()
     endif()
 endif()
