diff --git a/ext/libav/gstavauddec.c b/ext/libav/gstavauddec.c
index 7e35e2d..2baca46 100644
--- a/ext/libav/gstavauddec.c
+++ b/ext/libav/gstavauddec.c
@@ -151,6 +151,7 @@ gst_ffmpegauddec_init (GstFFMpegAudDec * ffmpegdec)
   ffmpegdec->context = avcodec_alloc_context3 (klass->in_plugin);
   ffmpegdec->context->opaque = ffmpegdec;
   ffmpegdec->opened = FALSE;
+  ffmpegdec->cmeta = NULL;
 
   ffmpegdec->frame = av_frame_alloc ();
 
@@ -577,6 +578,33 @@ gst_ffmpegauddec_frame (GstFFMpegAudDec * ffmpegdec, GstFlowReturn * ret)
 
   if (outbuf) {
     GST_LOG_OBJECT (ffmpegdec, "Decoded data, buffer %" GST_PTR_FORMAT, outbuf);
+    gsize osize = 0;
+    gsize size = 0;
+    GstAudioMeta *meta = gst_buffer_get_audio_meta (outbuf);
+    osize = meta ? meta->samples : (gst_buffer_get_size (outbuf) / ffmpegdec->info.bpf);
+
+    if (ffmpegdec->cmeta) {
+      GstAudioClippingMeta *cmeta = ffmpegdec->cmeta;
+      GST_LOG_OBJECT (ffmpegdec, "cmeta start %"GST_TIME_FORMAT" end %"GST_TIME_FORMAT,
+          GST_TIME_ARGS (cmeta->start), GST_TIME_ARGS(cmeta->end));
+      if (cmeta->format == GST_FORMAT_TIME) {
+        cmeta->start  = gst_util_uint64_scale(cmeta->start, ffmpegdec->info.rate, GST_SECOND);
+        cmeta->end = gst_util_uint64_scale(cmeta->end, ffmpegdec->info.rate, GST_SECOND);
+      }
+
+      if ((cmeta->start + cmeta->end) < osize) {
+        size = osize - cmeta->start - cmeta->end;
+        outbuf = gst_audio_buffer_truncate (outbuf, ffmpegdec->info.bpf, cmeta->start, size);
+        GST_LOG_OBJECT (ffmpegdec, "cmeta start %lld, end %lld, size %d, osize %d", cmeta->start, cmeta->end, size, osize);
+      } else {
+        /* drop the entire frame */
+        gst_buffer_unref(outbuf);
+        outbuf = NULL;
+        GST_LOG_OBJECT (ffmpegdec, "drop entire frame");
+      }
+    }
+
+    if (outbuf)
     *ret =
         gst_audio_decoder_finish_subframe (GST_AUDIO_DECODER_CAST (ffmpegdec),
         outbuf);
@@ -669,6 +697,7 @@ gst_ffmpegauddec_handle_frame (GstAudioDecoder * decoder, GstBuffer * inbuf)
     return gst_ffmpegauddec_drain (ffmpegdec, FALSE);
   }
 
+  ffmpegdec->cmeta = gst_buffer_get_audio_clipping_meta (inbuf);
   inbuf = gst_buffer_ref (inbuf);
   is_header = GST_BUFFER_FLAG_IS_SET (inbuf, GST_BUFFER_FLAG_HEADER);
 
diff --git a/ext/libav/gstavauddec.h b/ext/libav/gstavauddec.h
index d91de0d..87566c8 100644
--- a/ext/libav/gstavauddec.h
+++ b/ext/libav/gstavauddec.h
@@ -48,6 +48,7 @@ struct _GstFFMpegAudDec
   GstAudioInfo info;
   GstAudioChannelPosition ffmpeg_layout[64];
   gboolean needs_reorder;
+  GstAudioClippingMeta *cmeta;
 };
 
 typedef struct _GstFFMpegAudDecClass GstFFMpegAudDecClass;
-- 
2.32.0
